include/rdma/fi_errno.h

```diff
@@ -177,18 +177,22 @@ extern "C" {
 //#define	FI_ENOTRECOVERABLE	ENOTRECOVERABLE	/* State not recoverable */
 
 /* FI specific return values: >= 256 */
+#define FI_ERRNO_OFFSET	256
 
-#define FI_EOTHER		256		/* Unspecified error */
-#define FI_ETOOSMALL		257		/* Provided buffer is too small */
-#define FI_EOPBADSTATE		258		/* Operation not permitted in current state */
-#define FI_EAVAIL		259		/* Error available */
-#define FI_EBADFLAGS		260		/* Flags not supported */
-#define FI_ENOEQ		261		/* Missing or unavailable event queue */
-#define FI_EDOMAIN		262		/* Invalid resource domain */
-#define FI_ENOCQ		263		/* Missing or unavailable completion queue */
-#define FI_ECRC			264		/* CRC error */
-#define FI_ETRUNC		265		/* Truncation error */
-#define FI_ENOKEY		266		/* Required key not available */
+enum {
+	FI_EOTHER = FI_ERRNO_OFFSET, /* Unspecified error */
+	FI_ETOOSMALL,                /* Provided buffer is too small */
+	FI_EOPBADSTATE,              /* Operation not permitted in current state */
+	FI_EAVAIL,                   /* Error available */
+	FI_EBADFLAGS,                /* Flags not supported */
+	FI_ENOEQ,                    /* Missing or unavailable event queue */
+	FI_EDOMAIN,                  /* Invalid resource domain */
+	FI_ENOCQ,                    /* Missing or unavailable completion queue */
+	FI_ECRC,                     /* CRC error */
+	FI_ETRUNC,                   /* Truncation error */
+	FI_ENOKEY,                   /* Required key not available */
+	FI_ERRNO_MAX
+};
```

Comment:
I often find that the only information I've given when debugging is the actual error number, not string.  Trying to map the value back to the error type is a pain with an enum.  What if we specified each value?  I.e.:

enum {
   FI_EOTHER = 256,
   FI_ETOOSMALL = 257,
   ...
   FI_ENOKEY = 266,
   FI_ERRNO_MAX
};

These values cannot change once we've defined them.  We need to make sure that someone inserting a value in the enum doesn't shift the values down and break things.